### ISSUE TITLE
mathml2chtml fails with a newline at the end.

### ISSUE THREAD

**Issue by NSoiffer** [mathjax/MathJax#2805]
### Issue Summary

A newline at the end of some MathML in a call to mathml2chtml causes an error (actually, the call never returns)
This works
```
MathJax.mathml2chtml("<math><mi>XXX</mi></math>");
```
This fails:
```
MathJax.mathml2chtml("<math><mi>XXX</mi></math>\n");
```

### Steps to Reproduce:
I tried to get a codepen example to run, and codepen complained about "MathJax" not being defined. Here's some HTML that shows the problem (the "AFTER..." is never shown):
```
<!DOCTYPE html>
<html>
<head>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/mml-chtml.js"></script> 
<script>
function ConvertMathML() {
  MathJax.startup.defaultReady();
  document.getElementById("before").innerHTML = "BEFORE conversion";
  let foo = MathJax.mathml2chtml("<math><mi>XXX</mi></math>\n");
  document.getElementById("after").innerHTML = "AFTER conversion";
}
</script>
</head>
<body>
<button onclick="ConvertMathML()"> Click me to see bug </button>
<p id="before"></p>
<p id="after"></p>
</body>
</html>
```

**Comment by dpvc** [comment]
The reason this is happening is that MathJax parses the string you pass it as HTML, not XML (mainly to support embedded HTML in `<mtext>` elements, which we plan to support in the future).  That means your string produces two HTML nodes:  the `<math>` node followed by a `#text` node containing the new line, and MathJax complains about that.  If you check the browser console, you will see an error message to that effect.

You can set what type of parsing will be used, however, in the options for the MathML input jax.  

``` javascript
MathJax = {
  mml: {
    parseAs: 'xml'
  }
};
```

Unfortunately, there is a bug with the XML parsing that causes this to fail (it will be fixed in the next release).  But you can work around it using

``` javascript
MathJax = {
  mml: {
    parseAs: 'xml'
  },
  startup: {
    ready() {
      const {HTMLAdaptor} = MathJax._.adaptors.HTMLAdaptor;
      HTMLAdaptor.prototype.body = function(doc) {return doc.body || doc};
      MathJax.startup.defaultReady();
    }
  }
}
```

for now.

It would be reasonable for MathJax to trim the whitespace from the MathML before parsing it, however.

**Comment by NSoiffer** [comment]
Trimming whitespace was the solution I did to avoid the problem. It just took a while to figure out. I think MathJax trimming whitespace makes a lot of sense... but I don't know enough to know if there is any downside.

Unfortunately the message "MathML must consist of a single element" wasn't helpful in figuring out the problem in that as far as I could tell, I was sending a single element. I thought that maybe there was bad tagging in the MathML I generated.

**Comment by dpvc** [comment]
Yes, the message could probably be better.  But trimming the white space should be sufficient to resolve the main issue.
